piptools/scripts/compile.py

```diff
@@ -340,6 +356,7 @@ def cli(
     emit_index_url: bool,
     emit_options: bool,
     unsafe_package: tuple[str, ...],
+    config: str | None,
```

Comment:
Let's make this accept an already normalized `pathlib.Path` object:
```suggestion
    config: Path | None,
```